packages/react-reconciler/src/ReactFiberWorkLoop.old.js

```diff
@@ -2091,6 +2106,15 @@ function commitRootImpl(root, renderPriorityLevel) {
   // additional work on this root is scheduled.
   ensureRootIsScheduled(root, now());
 
+  if (recoverableErrors !== null) {
```

Comment:
The timing of these are interesting given that we were talking about logging order. If it's uncaught it'll be logged after this, but if it's caught the logging will happen in an error boundary who is supposed to log it in componentDidCatch which is a layout effect.

So it seems to me that this needs to be logged before the layout effect phase.

Reply:
That's a good point, though I don't know if we can be sure of the order unless we start tracking the recoverable errors per tree. If there were recoverable errors and regular errors (`componentDidCatch`) in the same commit, the current implementation can't tell which happened first because the recoverable error array is shared across the whole tree.

Reply:
Yea but if they're separate subtrees they're more likely independent and we could've rendered them in any order.

The cases I'm mostly thinking about is when you have something like a hydration that succeeded and then errored because it wasn't quite fixed. However, I can't think of a case where that wouldn't be two separate consecutive commits.